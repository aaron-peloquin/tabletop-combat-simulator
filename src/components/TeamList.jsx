import {
  Button,
  Card,
  CardActions,
  CardContent,
  Grid,
  Typography,

} from "@material-ui/core/"
// import { withStyles } from "@material-ui/core/styles"
import { connect } from "react-redux"

import SetEditCreature from "./../store/action/SetEditCreature"
import DeleteCreature from "./../store/action/DeleteCreature"

const TeamList = (Props) => {
  const {
    /** Actions */
    SetEditCreature,
    DeleteCreature,

    /** Attributes */
    Creatures,
    Team,
  } = Props
  return <Grid container>
    <Grid item xs={12}>Team {Team.toUpperCase()}</Grid>
    {Creatures.map((Creature, k)=>{
      return <Grid item md={6} key={k}>
        <Card>
          <CardContent>
            <Typography variant="body2">{Creature.name}</Typography>
          </CardContent>
          <CardActions>
            <Button onClick={() => { SetEditCreature(Creature) }}>Edit</Button>
            <Button onClick={() => { DeleteCreature(Creature.hash) }}>Delete</Button>
          </CardActions>
        </Card>
      </Grid>
    })}
  </Grid>
}

const MapStateToProps = (state, props) => {
  const { Team } = props
  const Creatures = state.creatures.filter((Creature)=>{
    return Creature.team === Team
  })
  return {
    Creatures,
  }
}

const MapActionsToProps = (dispatch) => {
  return {
    SetEditCreature: (Creature) => { SetEditCreature(dispatch, Creature) },
    DeleteCreature: (Hash) => { DeleteCreature(dispatch, Hash) },
  }
}

export default connect(MapStateToProps, MapActionsToProps)(TeamList)
