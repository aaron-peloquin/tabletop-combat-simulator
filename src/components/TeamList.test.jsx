import {shallow} from "enzyme"

import {TeamList} from "./TeamList"

jest.mock('./TeamListCreature')

describe("<TeamList />", () => {
  let props; let TeamListComponent

  beforeEach(() => {
    props = {
      Team: "a",
      Creatures: [
        { name: "foo" }
      ],
    }
    TeamListComponent = shallow(<TeamList {...props} />)
  })

  it("renders", () => {
    expect(typeof TeamListComponent).toBe("object")
  })

  it("snapshots", () => {
    expect(TeamListComponent.debug()).toMatchSnapshot()
  })
})
