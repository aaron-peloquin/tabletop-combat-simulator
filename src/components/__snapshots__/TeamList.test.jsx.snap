// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeamList /> snapshots 1`] = `
"<WithStyles(Grid) data-cy=\\"TeamList\\" data-team=\\"a\\" container={true} spacing={24} justify=\\"center\\">
  <WithStyles(Grid) item={true} xs={12}>
    <WithStyles(Typography) variant=\\"h4\\">
      Team 
      A
    </WithStyles(Typography)>
  </WithStyles(Grid)>
  <WithStyles(Grid) item={true} xs={12}>
    <WithStyles(Typography) variant=\\"body2\\">
      1
       Creatures
    </WithStyles(Typography)>
  </WithStyles(Grid)>
  <WithStyles(Grid) item={true} lg={6}>
    <Connect(WithStyles(RawTeamListCreature)) Creature={{...}} />
  </WithStyles(Grid)>
</WithStyles(Grid)>"
`;
